### ISSUE TITLE
CC has trouble with single digit MIDI messages and lots of MIDI reset messages

### ISSUE THREAD

**Issue by bwhitman** [shorepine/tulipcc#318]
User on discord reports that Tulip CC isn't working properly on Reaper

**Comment by bwhitman** [comment]
This is a fun one. On Desktop we ensure 3 byte midi messages, so for things like Reaper sending a clock message, it's just ignored. But on CC we parse these as they come in and the python `midi.py` parser was erroring out getting a single byte message. We'll fix that. 

But, also, by default Reaper is sending quite a lot of reset messages on transport play: sustain, PB and note off, to every channel. which is overloading the 48 byte UART buffer on tulip CC. I'll play with these sizes and make this stable.

**Comment by bwhitman** [comment]
OK, we've made a few fixes to the hardware tulip MIDI parser to make this work better. I'm now able to play reaper tracks to tulip CC just fine now. 

HOWEVER, i think we've found a bug in reaper itself, so watch out. For hardware MIDI ports, at least on my system, Reaper does NOT send note on messages if you have "Send clock to this device" ticked on (even if "Enable output to this device" is on.) I've verified this by watching the raw midi stream, and also plugging reaper directly into an off the shelf hardware synth and seeing it break there too. Reaper DOES send note ons alongside clocks over my Mac's IAC bus. so it's a very confusing bug. I'll look more into this with a logic analyzer soon but I do not think this is a tulip issue!
